### ISSUE TITLE
Cloned functions may still use source schema objects.

### ISSUE THREAD

**Issue by guignonv** [denishpatel/pg-clone-schema#57]
This issue is related to #34 and #35 fix.

For the "create functions" part, there is now:
`EXECUTE 'SET search_path = ' || quote_ident(source_schema);` to fix those issues.
However, I have a schema to clone with a function referencing a schema table without its schema prefix. The cloned function references the source schema table instead of the cloned one.
The `SELECT replace(qry, quote_ident(source_schema) || '.', quote_ident(dest_schema) || '.') INTO dest_qry;` would not help as the table is not prefixed in the function definition.

So, for instance, in the source schema "srcschema":
```
CREATE FUNCTION somefunc(bigint) RETURNS SETOF sometable
    LANGUAGE sql
    AS $_$SELECT sometable.* FROM sometable WHERE sometable.somefield > $1 AND  sometable.somefield < 2*$1 $_$;
```
But in the cloned schema I got:
```
CREATE FUNCTION somefunc(bigint) RETURNS SETOF srcschema.sometable
    LANGUAGE sql
    AS $_$SELECT sometable.* FROM srcschema.sometable WHERE sometable.somefield > $1 AND  sometable.somefield < 2*$1 $_$;
```
I guessed it is because the search path is set to the source schema and the object "sometable" is found there when the table is created. I think we should use the destination schema instead:
```EXECUTE 'SET search_path = ' || quote_ident(dest_schema);```
BUT the problem would be that some functions not cloned yet may be requested by others being created. I would see 2 ways of solving that issue: the first one would be to have a complex algorithm that would determine function dependencies and order them. The second one would be to have a stack of functions to create, start from the top of the stack, catch "missing function" exception while create each function and put back not created functions at the bottom of the stack. There should be some counter that prevents a function to be put at the bottom of the stack more than a certain number of times (20?) to avoid an infinite loop on the stack.
...or another approach I don't think of. :)

**Comment by guignonv** [comment]
Yet another idea to solve the function dependency problem: create all functions as prototypes in a first pass and then alter them with the full definition.

**Comment by MichaelDBA** [comment]
This is a bit too complicated to attack at the moment.  I think instead I will document this as a limitation and assumption going forward that schema-qualified references in functions will not clone successfully.

**Comment by guignonv** [comment]
This should be fixed by PR #58. I've implemented the approach of using prototype functions (functions with and empty body but with their final "signature") so they can be used anywhere and then updated with their full body without loosing any reference.
